src/main/java/com/zoo/animals/Animal.java

```diff
@@ -1,54 +1,55 @@
 package com.zoo.animals;
 
-public class Animal {
+import com.zoo.interfaces.IEatable;
+import com.zoo.interfaces.ISleepable;
+
+public abstract class Animal implements IEatable, ISleepable {
     //Protected and private fields
  protected String name;
- protected int age;
- protected double weight;
  private int energyLevel;
+
  //Constructor
-public Animal(String name, int age, double weight, int energyLevel) {
+public Animal(String name, int energyLevel) {
     this.name = name;
-    this.age = age;
-    this.weight = weight;
-    this.energyLevel = energyLevel;
+    this.energyLevel = 100;
```

Comment:
Why don't we use the default value of a field for this? :)

Reply:
Hint: imagine in the future you'll get 10 more overloads of this constructor

From some moment, it will be so easy to forget to set this field, and someone will forget, which leads to inconsistency - the object has an unexpectedly different state when a particular version of the constructor was used for its creation